iceoryx_utils/include/iceoryx_utils/cxx/timer.hpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_UTILS_PERIODIC_TIMER_HPP
+#define IOX_UTILS_PERIODIC_TIMER_HPP
+
+#include "iceoryx_utils/cxx/expected.hpp"
+#include "iceoryx_utils/internal/units/duration.hpp"
+#include "iceoryx_utils/posix_wrapper/semaphore.hpp"
+#include <thread>
+
+using namespace iox::units;
+using namespace iox::units::duration_literals;
```

Comment:
`using namespace` in *.hpp's leads to [namespace pollution](https://stackoverflow.com/questions/22903542/what-is-namespace-pollution) and higher build times. Better call `fromSeconds` et al:

```
  private:
    iox::units::Duration m_interval{units::Duration::fromSeconds(0)};
    iox::units::Duration m_timeForNextActivation{units::Duration::fromSeconds(0)};
```

`using namespace iox::units::duration_literals` is perfectly fine to use in `timer.cpp` in a local scope.